plugins/modules/ec2_vpc_net_info.py

```diff
@@ -181,57 +181,33 @@ def describe_vpcs(connection, module):
-        module.fail_json_aws(e, msg='Unable to describe if ClassicLinkDns is supported')
-
+    cl_enabled = {}
+    cl_dns_support = {}
     # Loop through the results and add the other VPC attributes we gathered
     for vpc in response['Vpcs']:
-        error_message = "Unable to describe VPC attribute {0}"
-        # We have to make two separate calls per VPC to get these attributes.
-        try:
-            dns_support = connection.describe_vpc_attribute(VpcId=vpc['VpcId'],
-                                                            Attribute='enableDnsSupport', aws_retry=True)
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg=error_message.format('enableDnsSupport'))
-        try:
-            dns_hostnames = connection.describe_vpc_attribute(VpcId=vpc['VpcId'],
-                                                              Attribute='enableDnsHostnames', aws_retry=True)
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg=error_message.format('enableDnsHostnames'))
-
-        # loop through the ClassicLink Enabled results and add the value for the correct VPC
-        for item in cl_enabled['Vpcs']:
-            if vpc['VpcId'] == item['VpcId']:
-                vpc['ClassicLinkEnabled'] = item['ClassicLinkEnabled']
-
-        # loop through the ClassicLink DNS support results and add the value for the correct VPC
-        for item in cl_dns_support['Vpcs']:
-            if vpc['VpcId'] == item['VpcId']:
-                vpc['ClassicLinkDnsSupported'] = item['ClassicLinkDnsSupported']
+        error_message = "Unable to describe VPC attribute {0} on VPC {1}"
+        cl_enabled = describe_classic_links(module, connection, vpc['VpcId'], 'ClassicLinkEnabled', error_message)
+        cl_dns_support = describe_classic_links(module, connection, vpc['VpcId'], 'ClassicLinkDnsSupported', error_message)
+        dns_support = describe_vpc_attribute(module, connection, vpc['VpcId'], 'enableDnsSupport', error_message)
+        dns_hostnames = describe_vpc_attribute(module, connection, vpc['VpcId'], 'enableDnsHostnames', error_message)
+        if cl_enabled:
+            # loop through the ClassicLink Enabled results and add the value for the correct VPC
+            for item in cl_enabled['Vpcs']:
+                if vpc['VpcId'] == item['VpcId']:
+                    vpc['ClassicLinkEnabled'] = item['ClassicLinkEnabled']
+        if cl_dns_support:
+            # loop through the ClassicLink DNS support results and add the value for the correct VPC
+            for item in cl_dns_support['Vpcs']:
+                if vpc['VpcId'] == item['VpcId']:
+                    vpc['ClassicLinkDnsSupported'] = item['ClassicLinkDnsSupported']
 
         # add the two DNS attributes
         vpc['EnableDnsSupport'] = dns_support['EnableDnsSupport'].get('Value')
```

Comment:
dns_support and dns_hostnames can (in theory) both now be empty, this will need to be handled.